packages/faucet/src/api/webserver.ts

```diff
@@ -73,6 +88,8 @@ export class Webserver {
 
           try {
             await faucet.credit(address, matchingDenom);
+            // Count addresses to prevent draining
+            this.addressCounter.push({ address: address, date: Date.now() });
```

Comment:
This adds the same address multiple times to the array. But `const addressUsed = this.addressCounter.find((x) => x.address === address);` will only search for the first entry. So after 24h, no request will be blocked anymore.

We probably make our lives easier using a Map instead.